app/builders/messages/message_builder.rb

```diff
@@ -49,10 +49,10 @@ def process_emails
     return unless @conversation.inbox&.inbox_type == 'Email'
 
     cc_emails = []
-    cc_emails = @params[:cc_emails].split(',') if @params[:cc_emails].present?
+    cc_emails = @params[:cc_emails].strip.split(/\s*,\s*/) if @params[:cc_emails].present?
```

Comment:
```suggestion
    cc_emails = @params[:cc_emails].gsub(/\s+/, '').split(',') if @params[:cc_emails].present?
```

Reply:
Just checking: Is it safe to eliminate all spaces here? Are these values ever in a format like `"John Doe" <john@doe.com>`?  Also I feel like I've heard that something like `"john doe"@doe.com` is a valid email address?

I wonder if these values should be sent in JSON format or something instead of being `join`ed with whitespace?

Reply:
@jordan-brough  You are right about the spaces being allowed in email  ref: https://stackoverflow.com/questions/2049502/what-characters-are-allowed-in-an-email-address

Generally, the approach we have been following in Chatwoot for email validation is based on the addresses supported by Gmail. ie (letters numbers and + ). We could take it as a separate effort to see whether we want to support all cases per RFC.

For this particular case

- We currently expect an array of coma-separated emails which has  alphanumeric characters 
- We don't support the format with the name in the email ex:  "John Doe<john@doe.com>"

---

I wonder if these values should be sent serialized in JSON format or something instead of being joined with commas? 

We will have to do some work for backward compatibility for the change. There will be older messages in the database with cc emails and bcc emails in the current format. We can go ahead with the current fix and take it up separately.